### PR TITLE
Filters work (in a simple form)

### PR DESCRIPTION
**Pull request by brevansio**
Lots of dirty code, but I got some filters working.

### CODE REVIEW

File: PRChecker/Shared/CommonViews/CheckBox.swift [hunk 1]
```diff
@@ -8,26 +8,26 @@
 import SwiftUI
 
 struct CheckBox: View {
-    let text: String?
+    @State var filter: Filter
```
Comment (Ckitakishi) [inline PRChecker/Shared/CommonViews/CheckBox.swift:11]: Checkbox does too many things, but I think it's acceptable for now lol

File: PRChecker/Shared/ViewModels/FilterViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                Filter(name: "Commented") { $0.viewerStatus == .commented },
+                Filter(name: "Approved") { $0.viewerStatus == .approved },
+            ]
+        )
+    }()
+    
+    lazy var sections: [FilterSection] = [
+        Self.statusFilter,
+        Self.reviewStatusFilter,
+        // TODO: Label Filter
```
Comment (Ckitakishi) [inline PRChecker/Shared/ViewModels/FilterViewModel.swift:37]: Hope we can filter by repository name and read/unread status too in the future

**Review by Ckitakishi — APPROVED** [review]

File: PRChecker/Shared/ViewModels/FilterViewModel.swift (hunk 2 above)
Comment (brevansio) [reply PRChecker/Shared/ViewModels/FilterViewModel.swift:37]: Yep. I was able to get the labels working, and the process is the same, so it should be possible to filter on anything in the PullRequestCell

Comment (Ckitakishi) [reply PRChecker/Shared/ViewModels/FilterViewModel.swift:37]: Nice! Let me merge this PR first.

**Pull request merged**